### ISSUE TITLE
Salesforce plugin customer requests

### ISSUE THREAD

**Issue by lharries** [PostHog/apps#11]
An ICP [customer](https://posthog.slack.com/archives/C03JJQT7G5D) has requested the following three things:

1. Being able to update more than 1 salesforce object with different sets of properties.
  - Currently we only support one salesforce object the (Path of the url where events will go to. No leading forward slash) combined with (Properties to include)
  - A potentially related problem: a number of apps often need a variable number of a set of fields e.g. the user interview needing to accept multiple feature flags and corresponding booking links. Currently you need to enter a comma separated list e.g. `feature_flag=booking_url,feature_flag_2=booking_url_2` however this isn't very friendly and ends up significantly more complicated for the salesforce example.
  - A potential solution would be to extend the plugin.json config to take a variable number of field groups
2. Ability to load the different salesforce object resource URLs into the app as a dropdown
  - More broadly, I think we want a way to fetch data from their server and present it as a dropdown in the config
3. A more non-dev friendly app setup and config process

I think we should prioritize (1) for now. (2) and (3) are less blocking for them and we can revisit during the Q2 work on the CDP

**Comment by simfish85** [comment]
This is an accurate summary.  The only thing I'd add here is that Salesforce allows users to create [Custom Objects](https://trailhead.salesforce.com/content/learn/modules/data_modeling/objects_intro) as well as the Standard Ones (Opportunity, Contact etc) so we need to have the ability to map events to both custom and standard ones, as well as the properties therein.

**Comment by pauldambra** [comment]
It's a shame you can't install multiple copies of one app with different configs :)

So currently you can set zero or more events as sources, and a single salesforce URL/properties list pair to be sent when those events are seen

<img width="510" alt="Screenshot 2023-01-17 at 19 48 41" src="https://user-images.githubusercontent.com/984817/212997667-0ff73f80-500b-4665-9427-d58a8492f4ee.png">

The request is to be able to set a set of zero or more events as sources and in reaction you send any/all of those events to multiple salesforce urls?

Or do we want a mapping

```
{
    "$pageview": {
       "properties": ["device_id", "etc"],
       "target": salesforce_url_one
    },
    "customerCheckedOut": {
       "properties": ["foo", "etc"],
       "target": salesforce_url_two
    },
    "etc": {
        "and": "so on"
    }
}
```

I've never used salesforce in anger so not sure what would be best

**Comment by simfish85** [comment]
Yes more of a mapping. Essentially a selected event(s) in PostHog is mapped to a record (URL) in SF eg:

user signed up -> Lead
insight analyzed -> Engagement

**Comment by pauldambra** [comment]
Had a run at it here https://github.com/Vinovest/posthog-salesforce/pull/31

I'm not really sure how to properly test this since I've no access to any salesforce 

I noticed that there is config for a property allowlist but the app doesn't filter properties at the moment, so I added that to the new mechanism

I've tried to leave the existing code in place

**Comment by pauldambra** [comment]
https://github.com/PostHog/salesforce-plugin/pull/8

Has been released, it allows sending different events to different paths by config
